File: agent/src/helpers/format-docker-uptime.ts
export function formatDockerUptime(status: string): string {
  const normalized = status.replace("Up ", "").replace("(healthy)", "").trim();

  const matches = normalized.match(
    /(\d+)\s(day|days|hour|hours|minute|minutes)/g,
  );

  if (!matches) {
    return normalized;
  }

  let days = 0;
  let hours = 0;
  let minutes = 0;

  matches.forEach((match) => {
    const [valueStr, unit] = match.split(" ");

    const value = Number(valueStr);

    if (unit.startsWith("day")) {
      days = value;
    }

    if (unit.startsWith("hour")) {
      hours = value;
    }

    if (unit.startsWith("minute")) {
      minutes = value;
    }
  });

  const result = [];

  if (days > 0) result.push(`${days}d`);
  if (hours > 0) result.push(`${hours}h`);
  if (minutes > 0) result.push(`${minutes}m`);

  return result.join(" ");
}

File: agent-go/helpers/format_docker_uptime.go
package helpers

import (
	"fmt"
	"regexp"
	"strconv"
	"strings"
)

func FormatDockerUptime(status string) string {
	normalized := strings.ReplaceAll(status, "Up ", "")
	normalized = strings.ReplaceAll(normalized, "(healthy)", "")
	normalized = strings.TrimSpace(normalized)

	re := regexp.MustCompile(`(\d+)\s(day|days|hour|hours|minute|minutes)`)
	matches := re.FindAllStringSubmatch(normalized, -1)

	if len(matches) == 0 {
		return normalized
	}

	result := []string{}

	for _, match := range matches {
		value, _ := strconv.Atoi(match[1])
		unit := match[2]

		switch {
		case strings.HasPrefix(unit, "day"):
			result = append(result, fmt.Sprintf("%dd", value))

		case strings.HasPrefix(unit, "hour"):
			result = append(result, fmt.Sprintf("%dh", value))

		case strings.HasPrefix(unit, "minute"):
			result = append(result, fmt.Sprintf("%dm", value))
		}
	}

	return strings.Join(result, " ")
}